### DISCUSSION TITLE
Will providing the valid format externally allow to import the domains?

### DISCUSSION THREAD

**Discussion by mzfr** [Findomain/Findomain#137] [Q&A] [answered]
Hey,

So I noticed that to import subdomains in the DB the findomain run following command:

```
findomain -t "TARGET_NAME" --import-subdomains "EXTERNAL_SUBDOMAIN_FILE" -m -q --http-status -i --pscan --iport 1 --lport 65537 --threads 150
```

which will return the output like:

```
TARGET_NAME, TARGET_IP, TARGET_URL, [OPEN PORTS]
```

So I wanted to ask will the import functionality work if I just provide a file with all the information in that format and just run the command like:

```
findomain --import-subdomains "INPUT_FILE_NAME"
```

I understand that the data in the `INPUT_FILE_NAME` should be something like:

```
TARGET_NAME, TARGET_IP, TARGET_URL, [OPEN PORTS]
```

because of the DB schema

**Comment by mzfr** [comment]
okay I don't want to open another threads so I'll just ask here, I notice if we run a port scan while importing external subdomains it immediately(within 2-3 sec) returns the open ports even when we have set ` --iport 1 --lport 65537`.

Now I tried to scan the same number of ports with `masscan`  or even `rustscan` it takes more than 10-15 minutes to do the full port scan. So I wanted to ask how does findomain do it so fast?

**Reply by Edu4rdSHL** [reply]
> okay I don't want to open another threads so I'll just ask here, I notice if we run a port scan while importing external subdomains it immediately(within 2-3 sec) returns the open ports even when we have set ` --iport 1 --lport 65537`.
> 
> Now I tried to scan the same number of ports with `masscan` or even `rustscan` it takes more than 10-15 minutes to do the full port scan. So I wanted to ask how does findomain do it so fast?

1) `65537` is an invalid port, maximum port number is `65535`
2) Findomain uses his own algorithm when doing ports scan avoiding scanning the same IP more than once, maybe that causes the difference. See https://github.com/Edu4rdSHL/unimap#how for details (Unimap was born from Findomain).

**Comment by Edu4rdSHL** [comment, marked as answer]
> Hey,
> 
> So I noticed that to import subdomains in the DB the findomain run following command:
> 
> ```
> findomain -t "TARGET_NAME" --import-subdomains "EXTERNAL_SUBDOMAIN_FILE" -m -q --http-status -i --pscan --iport 1 --lport 65537 --threads 150
> ```
> 
> which will return the output like:
> 
> ```
> TARGET_NAME, TARGET_IP, TARGET_URL, [OPEN PORTS]
> ```
> 
> So I wanted to ask will the import functionality work if I just provide a file with all the information in that format and just run the command like:
> 
> ```
> findomain --import-subdomains "INPUT_FILE_NAME"
> ```
> 
> I understand that the data in the `INPUT_FILE_NAME` should be something like:
> 
> ```
> TARGET_NAME, TARGET_IP, TARGET_URL, [OPEN PORTS]
> ```
> 
> because of the DB schema

Subdomains to import should have the following format:

```
subdomain1.tld
subdomain2.tld
subdomain3.tld
subdomain4.tld
subdomainN.tld
```
the additional fields for the database schema are automatically filled by Findomain according to the parameter given. If you put them in the `TARGET_NAME, TARGET_IP, TARGET_URL, [OPEN PORTS]` format, they're probably marked as invalid.
